src/AccessibilityInsights/MainWindow.xaml.cs

```diff
@@ -263,13 +264,17 @@ void InitPanes()
         private static void SupportDebugging()
         {
             var args = Environment.GetCommandLineArgs();
-
-            if (args.Contains(ArgAttach))
-            {
-                var dlg = new MessageDialog();
-                dlg.Message = Properties.Resources.SupportDebuggingDialogMessage;
-                dlg.ShowDialog();
-            }
+            Parser.Default.ParseArguments<CommandOptions>(args.Skip(1))
```

Comment:
>ParseArguments<CommandOptions>(args.Skip(1)) [](start = 27, length = 44)

Code smell: You're calling ParseArguments in 3 different places. You're also exposing the setters on CommandOptions. You can fix both by creating a wrapper class that _contains_ a CommandOptions class, initializes it once (maybe on-demand via a Lazy), then exposes get-only properties.

Reply:
Your comment in InstallationEngine brings up another reason to consolidate the parsing. We may want to place some limitations about when command line overrides are respected. Settings that are not intended to be used by end users could potentially be ignored unless some runtime condition is satisfied. Maybe an environment variable?

---
In reply to: [298772144](https://github.com/microsoft/accessibility-insights-windows/pull/407#discussion_r298772144) [](ancestors = 298772144)

Reply:
I agree with the below comment about removing the help text for our command line arguments--no need to advertise our command line features. I don't think we should take any active steps to make the arguments more difficult to use, though--is there are particular scenario you're concerned about?

Reply:
Chatted with Dave before heading out on Friday and we settled on what you've suggested, John - I will remove the HelpText, add `//comments`, but we won't take any other steps. Using these parameters in non-testing scenarios (except the file one, perhaps) feels like relying on undocumented behavior / implementation details